app/forms/prior_authority/steps/service_cost_form.rb

```diff
@@ -14,98 +14,27 @@ def initialize(attrs)
           attrs[:prior_authority_granted] = attrs[:application].prior_authority_granted
         end
 
-        # This will be nil when loaded on update, populated on edit
-        attrs[:service_type] ||= attrs[:record].service_type
         super(attrs)
       end
 
-      PER_ITEM = 'per_item'.freeze
-      PER_HOUR = 'per_hour'.freeze
-      VARIABLE = 'variable'.freeze
-      COST_TYPES = [PER_ITEM, PER_HOUR].freeze
-
-      attribute :service_type, :value_object, source: QuoteServices
-
       attribute :prior_authority_granted, :boolean
       attribute :ordered_by_court, :boolean
       attribute :related_to_post_mortem, :boolean
-      attribute :user_chosen_cost_type, :string
-      attribute :items, :integer
-      attribute :cost_per_item, :decimal, precision: 10, scale: 2
-      attribute :period, :time_period
-      attribute :cost_per_hour, :decimal, precision: 10, scale: 2
 
       validates :prior_authority_granted, inclusion: { in: [true, false], allow_nil: false }
       validates :ordered_by_court, inclusion: { in: [true, false], allow_nil: false }, if: :court_order_relevant
       validates :related_to_post_mortem, inclusion: { in: [true, false], allow_nil: false }, if: :post_mortem_relevant
-      validates :user_chosen_cost_type, inclusion: { in: COST_TYPES, allow_nil: false }, if: :variable_cost_type?
 
-      validates :items, presence: true, numericality: { greater_than: 0 }, if: :per_item?
-      validates :cost_per_item, presence: true, numericality: { greater_than: 0 }, if: :per_item?
-
-      validates :period, presence: true, time_period: true, if: :per_hour?
-      validates :cost_per_hour, presence: true, numericality: { greater_than: 0 }, if: :per_hour?
+      include QuoteCostValidations
```

Comment:
I hate this but don't have a better suggestion...

Reply:
Yup 😭